include/ensmallen_bits/callbacks/early_stop_at_min_loss.hpp

```diff
@@ -28,11 +33,33 @@ class EarlyStopAtMinLoss
    * @param patienceIn The number of epochs to wait after the minimum loss has
    *    been reached or no improvement has been made (Default: 10).
    */
-  EarlyStopAtMinLoss(const size_t patienceIn = 10) :
-      patience(patienceIn),
+  EarlyStopAtMinLoss<MatType>(
+      const size_t patienceIn = 10,
+      std::ostream& output = arma::get_cout_stream())
+    : callback_used(false), patience(patienceIn), 
       bestObjective(std::numeric_limits<double>::max()),
-      steps(0)
-  { /* Nothing to do here */ }
+      steps(0), output(output)
+  { /* Nothing to do here */
+  }
+
+  /**
+   * Set up the early stop at min loss class, which keeps track of the minimum
+   * loss and stops the optimization process if the loss stops decreasing.
+   *
+   * @param func, callback to return immediate loss calculated bt the networl
+   * @param patienceIn The number of epochs to wait after the minimum loss has
+   *    been reached or no improvement has been made (Default: 10).
+   */
+  EarlyStopAtMinLoss<MatType>(
+      std::function<double(const MatType&)> func,
+      const size_t patienceIn = 10,
+      std::ostream& output = arma::get_cout_stream())
+    : callback_used(true), 
+      patience(patienceIn), bestObjective(std::numeric_limits<double>::max()),
+      steps(0), output(output), localFunc(func)
```

Comment:
```suggestion
      patience(patienceIn),
      bestObjective(std::numeric_limits<double>::max()),
      steps(0),
      output(output),
      localFunc(func)
```
I think the code is clearer when these are on different lines. :+1: